src/unitxt/operators.py

```diff
@@ -1029,10 +1029,13 @@ def process(self, multi_stream: MultiStream) -> MultiStream:
             [*ele[0]] if isinstance(ele[0], tuple) else ele[0]
             for ele in values_and_counts
         ]
+        multi_stream_to_return = {}
         for name in multi_stream:
```

Comment:
First time seeing this, I think this might should be done in a lazy way like other MultiStreamOpertors?

Reply:
Hi @elronbandel ,
It seems to me that all MultiStreamOperator-s in operators.py return actual multi-stream-s, not a dictionary of generators. So I think that also `process` of this class has to un-lazy the individual streams.  And this is actually equivalent to what we currently have.

Reply:
I suggested a larger change in the issue by using a different base class:

```
class ExtractFieldValues(InstanceOperatorWithMultiStreamAccess):
    field: str
    stream_name: str
    overall_top_frequency_percent: Optional[int] = 100
    min_frequency_percent: Optional[int] = 0
    to_field: str
    process_every_value: Optional[bool] = False

    def prepare(self):
        self.local_cache = None

    def verify(self):
            return super().verify()

    def process(
        self, instance: Dict[str, object], multi_stream: MultiStream
    ) -> Dict[str, object]:
        try:
            if self.local_cache is None:
                self.local_cache =  calculate_extracted_values(multi stream)
         
            instance[self.to_field] =  self.local_cache
            return instance
        except Exception as e:
            raise Exception(
                f"Unable to fetch instances from '{self.source_stream}' to '{self.target_field}'"
            ) from e
  ````          
            
            
  The parent class uses the lazy scheme:
  ```

  class InstanceOperatorWithMultiStreamAccess(StreamingOperator):
    """A class representing an instance operator with global access in the streaming system.

    An instance operator with global access is a type of `StreamingOperator` that operates on individual instances within a `Stream` and can also access other streams.
    It uses the `accessible_streams` attribute to determine which other streams it has access to.
    In order to make this efficient and to avoid qudratic complexity, it caches the accessible streams by default.
    """

    def __call__(self, multi_stream: Optional[MultiStream] = None) -> MultiStream:
        result = {}

        for stream_name, stream in multi_stream.items():
            stream = Stream(
                self.generator,
                gen_kwargs={"stream": stream, "multi_stream": multi_stream},
            )
            result[stream_name] = stream

        return MultiStream(result)

    def generator(self, stream, multi_stream):
        yield from (self.process(instance, multi_stream) for instance in stream)

    @abstractmethod
    def process(self, instance: dict, multi_stream: MultiStream) -> dict:
        pass
```

Reply:
Thanks, @yoavkatz , still:  how did you realize there was a problem with the previous (now in main) implementation of ExtractFieldValue?
Please share so I can make it into an important unit test

Reply:
I happened during the use in a real use case.   I saw that the field which was presumably extracted was not there  there.  I'm not 100% sure how to recreate the scenario in unitest.

Reply:
This is the only reason to shake the current implementation (now in main). @elronbandel says it is OK streaming-lazyness-wise.
The whole field was missing? or the name was still there, but no value (being the list of most common values) in that field?

Reply:
The whole field was missing.   Maybe you can try to use it in a card, and actually make it work.  For example, in /Users/yoavkatz/migrate/v10/unitxt/prepare/cards/sst2.py replace 
```
AddFields(
            fields={
                "classes": ["negative", "positive"],
                ```
                
                With the ExtractFieldValues.

Reply:
@yoavkatz , @elronbandel , I committed a new version, per Elron's idea, which employs AddFields operator.  With it, it is easy to set deep_copy, which can help in the cases that Yoav ran into.

Reply:
@yoavkatz , @elronbandel , I committed yet another very lazy implementation, shamelessly adopted from the only relevant multistream->multistream sister class: `SplitByValue`.  I also commit a version of `sst2.py`, and the yielded card `sst2.json`, just for you to see: `sst2.py` runs all the way through (successfully completing test_card). That is the only sign that I can see from here that everything is alright. 
I suggest **not** to merge sst2 into main. This is just for you to see that the fields are not gone.

Reply:
well, test_recipe.py caught the change of order of `positive`, `negative` inside field `classes`.  I need to learn how the template is generated from classes. For now, I just allowed the test to pass.

Reply:
Thanks Dafna. I think we should keep the new sst2, because it's a good example.